ecllibrary/std/DataPatterns/Profile.ecl

```diff
@@ -598,90 +647,103 @@ EXPORT Profile(inFile,
+                                                                            #END,
+                                                    BOOLEAN             is_filled :=
+                                                                            #IF(%_IsSetType%(%'@type'%))
+                                                                                COUNT(_inFile.#EXPAND(%'namePrefix'% + %'@name'%)) > 0
+                                                                            #ELSEIF(REGEXFIND('(unicode)|(utf)', %'@type'%))
+                                                                                LENGTH(%_TrimmedUni%(_inFile.#EXPAND(%'namePrefix'% + %'@name'%))) > 0
+                                                                            #ELSEIF(REGEXFIND('string', %'@type'%))
+                                                                                LENGTH(%_TrimmedStr%(_inFile.#EXPAND(%'namePrefix'% + %'@name'%))) > 0
+                                                                            #ELSEIF(REGEXFIND('data', %'@type'%))
+                                                                                LENGTH(_inFile.#EXPAND(%'namePrefix'% + %'@name'%)) > 0
+                                                                            #ELSEIF(%'@type'% = 'boolean')
+                                                                                TRUE
+                                                                            #ELSE
+                                                                                _inFile.#EXPAND(%'namePrefix'% + %'@name'%) != 0
+                                                                            #END,
+                                                    BOOLEAN             is_number :=
+                                                                            #IF(%_IsSetType%(%'@type'%))
+                                                                                FALSE
+                                                                            #ELSEIF(REGEXFIND('(integer)|(unsigned)|(decimal)|(real)', %'@type'%))
+                                                                                TRUE
+                                                                            #ELSE
+                                                                                FALSE
+                                                                            #END
+                                                },
+                                                _inFile.#EXPAND(%'namePrefix'% + %'@name'%),
+                                                LOCAL
+                                            ),
+                                            TRANSFORM(%DataInfoRec%, SELF := LEFT)
+                                    ),
+                                DATASET
```

Comment:
Diff: This is the else part of an IF() above, handling the case where a child dataset contains no records anywhere in the outer dataset.